src/core/main.go

```diff
@@ -233,7 +233,25 @@ func main() {
 	if err != nil {
 		log.Warningf("oidc.FixEmptySubIss() errors out, error: %v", err)
 	}
-	systemartifact.ScheduleCleanupTask(ctx)
+	// Scheduling of system artifact depends on the jobservice, where gorountine is used to avoid the circular
+	// dependencies between core and jobservice.
+	go func() {
+		url := config.InternalJobServiceURL() + "/api/v1/stats"
+		checker := health.HTTPStatusCodeHealthChecker(http.MethodGet, url, nil, 60*time.Second, http.StatusOK)
+		options := []retry.Option{
+			retry.InitialInterval(time.Millisecond * 500),
+			retry.MaxInterval(time.Second * 10),
+			retry.Timeout(time.Minute),
+			retry.Callback(func(err error, sleep time.Duration) {
+				log.Debugf("failed to ping %s, retry after %s : %v", url, sleep, err)
+			}),
+		}
+		if err := retry.Retry(checker.Check, options...); err != nil {
+			log.Errorf("failed to check the jobservice health status: timeout, error: %v", err)
+			return
```

Comment:
It means if the jobservice not be ready in 1 minute,  that will not create schedule task? do we need log the information or fatal?